codice/ddf
# DDF-2119: As an Integrator, I want to be able to trigger the caching of selected products in order to allow users to download them from the cache at a later time

#### What does this PR do?
This PR adds a new ResourceDownloadEndpoint and an associated ActionProvider. The ResourceDownloadEndpoint only downloads products to the local cache and does not return the product to the cllient.  The DownloadResourceActionProvider only returns an action if caching is enabled and the product is not currently in the cache.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@garrettfreibott 
@oconnormi 
@Lambeaux 
@bantillo 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@coyotesqrl
@shaundmorris

#### How should this be tested?
Run PRBS and do the following manual test:
1) Start DDF
2) Setup a federated source
3) Upload some images to use a products
4) Run a wildcard query from the Search UI
5) Select a metacard
6) On the Actions tab, select "Download Resource to local cache"
7) Run a wildcard query from the Search UI again
8) Select the same metacard that was selected on step 5
9) On the Actions tab, verify that the link (Action) "Download Resource to local cache" does NOT exist. It doesn't exist because the product is already in the cache.

#### Any background context you want to provide?

#### What are the relevant tickets?
DDF-2119
#### Screenshots (if appropriate)
#### Checklist:
